scripts/coins-logon-widget.js

```diff
@@ -121,37 +85,83 @@ CoinsLogonWidget.prototype._setState = function() {
     this.on(EVENTS.LOGOUT_ERROR, this.onError);
     this.on(EVENTS.LOGOUT_SUCCESS, this.onLogout);
 
-    auth.isLoggedIn().then(function(isLoggedIn) {
-        if (isLoggedIn) {
-            self.emit(EVENTS.LOGIN_SUCCESS, {
-                username: auth.getUsername(),
-            });
-        }
-    });
-};
+    this.init();
+
+    // Perform an auth check if the option is set
+    if (this._options.checkAuth) {
+        auth.isLoggedIn().then(function(isLoggedIn) {
+            if (isLoggedIn) {
+                self.emit(EVENTS.LOGIN_SUCCESS, {
+                    username: auth.getUsername(),
+                });
+            }
+        });
+    }
+}
+
+/**
+ * Inherit from `EventEmitter`:
+ * {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Inheritance_and_the_prototype_chain#Example}
+ */
+CoinsLogonWidget.prototype = Object.create(EventEmitter.prototype);
+CoinsLogonWidget.prototype.constructor = CoinsLogonWidget;
+
+/* ==========================================================================
+   Actions
+   ========================================================================== */
 
 CoinsLogonWidget.prototype.destroy = function() {
-    this.form.destroy();
-    delete this.form;
+    this.update(null);
+
     this.removeAllListeners();
-    delete this.options;
+
+    delete this._options;
+    delete this._rootNode;
+    delete this._state;
+    delete this._tree;
```

Comment:
nah, it would be GC'd when whatever instantiates `myWidget` is no longer referenced by anything in memory.  so, a page change, obviously.  or `delete myWidget`.